src/index.js

```diff
@@ -467,6 +457,7 @@ class Enmap extends Map {
    * @returns {Enmap} The enmap.
    */
   push(key, val, path = null, allowDupes = false) {
+    this.#checkKeyOrArrayOfKeys(key);
     const data = this.get(key);
     this.#check(key, 'Array', path);
```

Comment:
Could we move the checkKeyOrArrayOfKeys to the check function?

Reply:
`this.#check` is rather for checking a property. I removed the `this.#checkKeyOrArrayOfKeys(key);` to get rid of the redundant cases